docs/how-to/create-epics-ioc-with-ca-and-pva.md

```diff
@@ -0,0 +1,951 @@
+   - Test each phase
+
+3. **Testing Checklist**:
+   - ✓ All PVs accessible via `caget`/`pvget`
+   - ✓ Write operations work via `caput`/`pvput`
+   - ✓ Update rates match requirements
+   - ✓ Error handling works correctly
+   - ✓ Existing control system scripts still work
+   - ✓ GUIs display correctly
+
+4. **Deployment**:
+   - Document PV name changes (if any)
+   - Update control system configuration
+   - Train operators on any differences
+   - Monitor for issues
+
+### Benefits of Migration
+
+- **Less boilerplate**: No separate database and device support files
+- **Python ecosystem**: Use standard Python tools and libraries
+- **Better testing**: Unit test controllers with standard Python tools
+- **Multi-protocol**: Serve CA, PVA, Tango, REST, GraphQL simultaneously
+- **Type safety**: Python type hints catch errors early
+- **Easier maintenance**: Single language (Python) for device logic
+
+## Troubleshooting
+
+### PVs Not Appearing
+
+**Problem**: `caget MY-DEVICE:Temperature` returns "Channel connect timed out"
+
+**Solutions**:
+- Check `EPICS_CA_ADDR_LIST` environment variable
+- Verify PV prefix matches (case-sensitive)
+- Ensure IOC is running: look for "Running IOC" in logs
+- Check firewall allows EPICS CA ports (5064-5065)
+
+### Connection Errors
+
+**Problem**: `ConnectionError` when IOC starts
+
+**Solutions**:
+- Verify device IP address and port are correct
+- Check network connectivity: `ping <device-ip>`
+- Ensure device is powered on and responding
+- Check device documentation for correct protocol
+
+### Slow Updates
+
+**Problem**: PVs update slower than expected
+
+**Solutions**:
+- Check `update_period` in `io_ref` (lower = faster updates)
+- Reduce number of attributes with fast updates
+- Optimize device communication (batch queries if possible)
+- Consider if device can handle higher query rate
+
+### Import Errors
+
+**Problem**: `ModuleNotFoundError: No module named 'p4p'` or similar
+
+**Solutions**:
+```bash
+# Reinstall with extras
+pip install --force-reinstall 'fastcs[ca,pva]'
+
+# Or install individually
+pip install pythonSoftIOC
+pip install p4p
+```
+
+### GUI Not Generating
+
+**Problem**: `.bob` file not created
+
+**Solutions**:
+- Check `output_path` is valid and writable
+- Ensure directory exists (or will be created)
+- Look for errors in IOC startup logs
+- Verify `gui` parameter is passed to transport
+
+## Best Practices
+
+### PV Naming Conventions
+
+```python
+# Good: Clear, hierarchical names
+MY-DEVICE:Temperature
+MY-DEVICE:PowerSupply:Voltage
+MY-DEVICE:Motor1:Position
+
+# Avoid: Abbreviations, unclear names
+MY-DEV:TEMP
+MY-DEVICE:PSU1_V
+MY-DEVICE:M1POS
+```
+
+### When to Use CA vs PVA
+
+**Use both** when:
+- Supporting legacy and modern clients
+- During transition periods
+- Maximum compatibility needed
+
+**Use CA only** when:
+- Only legacy clients exist
+- Proven stability required
+- Simple scalar values only
+
+**Use PVA only** when:
+- Only modern clients (Phoebus)
```

Comment:
It figured out that CA supports phoebus [before](https://github.com/DiamondLightSource/FastCS/pull/283/files#diff-0be7d3b7211cdf39e68f49bda810e7f92340598cd13a59bb4e74c15be1d21ac6R213)

Reply:
It suggests that Phoebus is PVA only